src/Oscoin/Storage/State.hs

```diff
@@ -1,62 +1,34 @@
--- | Implements general state storage for blockchains.
 module Oscoin.Storage.State
-    ( lookupState
-    , storeState
-    , storeStateIO
-    , withHandle
+    ( MonadStateStore(..)
     , StateStore
-    , Handle
-    , new
-    , fromState
-    , fromStateM
+    , lookupState
+    , storeState
     ) where
 
 import           Oscoin.Prelude
 
 import           Oscoin.Crypto.Blockchain.Block (StateHash)
-import           Oscoin.Crypto.Hash (Hash)
 import qualified Oscoin.Crypto.Hash as Crypto
+import           Oscoin.Storage.HashStore
+
+type StateStore c st m = HashStore c st m
+
+class Monad m => MonadStateStore c st m | m -> st, m -> c where
```

Comment:
Question: do we need to maintain this typeclass at all, after this patch? I thought one of the main objective of this PR was to get rid of this typeclass entirely. What am I missing? Perhaps this is just an intermedia step towards getting rid of it? 😉

Reply:
We are very close to getting rid of the type class but it still requires some effort and I wanted to keep the PR small. I’ll follow up on this after I also transition the receipt store.

Reply:
👌 👏